# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/XPCOMInit.cpp
# Commit: a3fd24f5f0a7
# Full Hash: a3fd24f5f0a7bc094409fdbdbe1ef92fc459677b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-23 21:49:00
# Regressor Bug: 1579367
# File Overlap Count: 1
# Description:
#   Bug 1579367 - Initialize XPCJSContext explicitly, after loading user prefs. r=kmag
#   
#   This way we get the correct values for start-up prefs in the parent process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D51061
# ==============================================================================

diff -r e06f2a4de969 -r a3fd24f5f0a7 xpcom/build/XPCOMInit.cpp
--- a/xpcom/build/XPCOMInit.cpp	Fri Nov 22 09:48:58 2019 +0000
+++ b/xpcom/build/XPCOMInit.cpp	Fri Nov 22 12:40:17 2019 +0000
@@ -248,7 +248,8 @@
 // Note that on OSX, aBinDirectory will point to .app/Contents/Resources/browser
 EXPORT_XPCOM_API(nsresult)
 NS_InitXPCOM(nsIServiceManager** aResult, nsIFile* aBinDirectory,
-             nsIDirectoryServiceProvider* aAppFileLocationProvider) {
+             nsIDirectoryServiceProvider* aAppFileLocationProvider,
+             bool aInitJSContext) {
   static bool sInitialized = false;
   if (sInitialized) {
     return NS_ERROR_FAILURE;
@@ -461,7 +462,6 @@
   // Init mozilla::SharedThreadPool (which needs the service manager).
   mozilla::SharedThreadPool::InitStatics();
 
-  mozilla::ScriptPreloader::GetSingleton();
   mozilla::scache::StartupCache::GetSingleton();
   mozilla::AvailableMemoryTracker::Init();
 
@@ -485,6 +485,10 @@
       loop->thread_name().c_str(), loop->transient_hang_timeout(),
       loop->permanent_hang_timeout());
 
+  if (aInitJSContext) {
+    xpc::InitializeJSContext();
+  }
+
   return NS_OK;
 }
 